### PR TITLE
Elastic search v0

### PR DESCRIPTION
**Pull request by DonHaul**

### CODE REVIEW

**Comment by drjova** [comment]
@DonHaul you were right, this is wrong https://github.com/inspirehep/backoffice/blob/main/.github/workflows/integration-tests.yml#L24 we should use the opensearch image from dockerhub, could you please make a separate PR with the correct image?

**Comment by DonHaul** [comment]
> @DonHaul you were right, this is wrong https://github.com/inspirehep/backoffice/blob/main/.github/workflows/integration-tests.yml#L24 we should use the opensearch image from dockerhub, could you please make a separate PR with the correct image?

Sure.
Just curious what's the reason we should not use the images in the cern registry?

**Comment by drjova** [comment]
> > @DonHaul you were right, this is wrong https://github.com/inspirehep/backoffice/blob/main/.github/workflows/integration-tests.yml#L24 we should use the opensearch image from dockerhub, could you please make a separate PR with the correct image?
> 
> Sure. Just curious what's the reason we should not use the images in the cern registry?

good question, for the following reasons:
1.  it's wrong, should be `inspirehep`
2. while testing we are using a different image

hence I will recommend to align these images, we can use in both cases the one from the registry: `registry.cern.ch/cern-sis/inspirehep/opensearch`

**Comment by DonHaul** [comment]
Okay done, regarding the local.yml environment variables I don't know exactly what they should be so I modify them to be a mix between what they were before and the ones found in github actions

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     runs-on: ubuntu-latest
     services:
       opensearch:
-        image: registry.cern.ch/cern-sis/inspire/opensearch
```
Comment (drjova) [inline .github/workflows/integration-tests.yml]: @DonHaul we have to make a separate PR for this change and merge it first.

File: backoffice/utils/pagination.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+from django_elasticsearch_dsl_drf.pagination import QueryFriendlyPageNumberPagination
+from rest_framework.pagination import PageNumberPagination
+
+
+class StandardResultsSetPagination(PageNumberPagination):
+    page_size = 10
+    page_size_query_param = "page_size"
```
Comment (drjova) [inline backoffice/utils/pagination.py]: I would leave it `size`

File: backoffice/utils/pagination.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    max_page_size = 100
+
+
+class OSStandardResultsSetPagination(QueryFriendlyPageNumberPagination):
+    page_size = 10
+    page_size_query_param = "page_size"
```
Comment (drjova) [inline backoffice/utils/pagination.py]: ditto

**Review by drjova — CHANGES_REQUESTED** [review]
Thanks, some minor changes

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (DonHaul) [reply .github/workflows/integration-tests.yml]: Separate PR here:
https://github.com/inspirehep/backoffice/pull/39/files

**Comment by drjova** [comment]
@DonHaul could you please rebase?

**Comment by drjova** [comment]
@DonHaul a test is failing

**Comment by DonHaul** [comment]
> @DonHaul a test is failing



> @DonHaul a test is failing



> @DonHaul a test is failing

test fixed
Added extra tests for the index search

**Review by drjova — APPROVED** [review]
Looks good, let's try it.
